src/Process.js

```diff
@@ -1,59 +1,66 @@
 // is blocking; if so, the amount of blocking time needed is
 // randomly determined.
 class Process {
-    constructor(pid, cpuTimeNeeded=null, blocking=false) {
-        this._pid = pid;
-        this.queue = null;
-        this.cpuTimeNeeded = (cpuTimeNeeded !== null) ? cpuTimeNeeded : Math.round(Math.random() * 1000);
-        this.blockingTimeNeeded = blocking ? Math.round(Math.random() * 100) : 0;
-        // A bool representing whether this process was toggled from blocking to non-blocking or vice versa
-        this.stateChanged = false;
-    }
-    
-    setParentQueue(queue) {
-
-    }
+  constructor(pid, cpuTimeNeeded = null, blocking = false) {
+    this._pid = pid
+    this.queue = null
+    this.cpuTimeNeeded =
+      cpuTimeNeeded !== null ? cpuTimeNeeded : Math.round(Math.random() * 1000)
+    this.blockingTimeNeeded = blocking ? Math.round(Math.random() * 100) : 0
+    // A bool representing whether this process was toggled from blocking to non-blocking or vice versa
+    this.stateChanged = false
+  }
 
-    isFinished() {
+  setParentQueue(queue) {
+    this.queue = queue
+  }
 
+  isFinished() {
+    if (!this.cpuTimeNeeded && !this.blockingTimeNeeded && !this.queue) {
+      return true
     }
+    return false
+  }
 
-    // If no blocking time is needed by this process, decrement the amount of 
-    // CPU time it needs by the input time
-    // If blocking time is needed by this process, move it to the blocking queue
-    // by emitting the appropriate interrupt
-    // Make sure the `stateChanged` flag is toggled appropriately
-    executeProcess(time) {
-
-   }
-
-   // If this process requires blocking time, decrement the amount of blocking
-   // time it needs by the input time
-   // Once it no longer needs to perform any blocking execution, move it to the 
-   // top running queue by emitting the appropriate interrupt
-   // Make sure the `stateChanged` flag is toggled appropriately
-    executeBlockingProcess(time) {
-
+  // If no blocking time is needed by this process, decrement the amount of
+  // CPU time it needs by the input time
+  // If blocking time is needed by this process, move it to the blocking queue
+  // by emitting the appropriate interrupt
+  // Make sure the `stateChanged` flag is toggled appropriately
+  executeProcess(time) {
+    if (!this.blockingTimeNeeded) {
+      this.cpuTimeNeeded - time
+    } else if (this.blockingTimeNeeded) {
+      this.stateChanged = !this.stateChanged
+      return SchedulerInterrupt.PROCESS_BLOCKED
     }
+  }
 
-    // Returns this process's stateChanged property
-    isStateChanged() {
+  // If this process requires blocking time, decrement the amount of blocking
+  // time it needs by the input time
+  // Once it no longer needs to perform any blocking execution, move it to the
+  // top running queue by emitting the appropriate interrupt
+  // Make sure the `stateChanged` flag is toggled appropriately
+  executeBlockingProcess(time) {}
```

Comment:
This is what I would tackle next - remember you will need to separate processes by whether or not they finish execution during their allotted time slice. And remember you have access to the other methods you have created!